### PR TITLE
Add inline API reference

### PR DESCRIPTION
**Pull request by tensiuyan**
connect #463

### CODE REVIEW

**Comment by tensiuyan** [comment]
@chpapa let me know if I have done it correctly. If yes I will move on to other guides. Thanks!

**Comment by chpapa** [comment]
@tensiuyan Looks great!

Only thing I changed is I added the API ref link for User object and whoami which I think is needed given the context and will improve readability.

Actually I don't mind if you keep working on each doc and I add what I saw might be missing in this way.

BTW, I have also switched some sample code positions, and added "What's next from here?" at the bottom, see if you think these make sense?

**Comment by tensiuyan** [comment]
@chpapa Good. From now on I will
- add api ref link
- fix obvious typos
- add what's next from here

**Comment by bensonby** [comment]
@chpapa @tensiuyan Seems like adding a link to an inline code doesn't style well. The user will not know it's a clickable link. How about making it like `skygear.currentUser` ([API doc](https://doc.esdoc.org/github.com/skygeario/skygear-SDK-JS/class/lib/container.js~Container.html#instance-get-currentUser)).

And please put all the URLs at the bottom of the markdown for easier link management - You can refer to the example at https://raw.githubusercontent.com/SkygearIO/skygear-doc/master/content/cloud-function/intro-and-deployment/python.md

File: content/auth/basics/js.md [hunk 1]
```diff
@@ -89,11 +89,11 @@ If there is an authenticated user, it will give you a user object like this:
 }
 ```
 
-Please be reminded that the `currentUser` object only retrieved when user
+Please be reminded that the [`currentUser`](https://doc.esdoc.org/github.com/skygeario/skygear-SDK-JS/class/lib/container.js~Container.html#instance-get-currentUser) object is only retrieved when user
 login / signup Skygear.
 
 To get the latest information (e.g. roles, emails, etc.) of the current user,
-you can ask "Who am I" to Skygear:
+you can ask call [`whoami()`](https://doc.esdoc.org/github.com/skygeario/skygear-SDK-JS/class/lib/container.js~Container.html#instance-method-whoami):
```
Comment (bensonby) [inline content/auth/basics/js.md]: @tensiuyan should be "You can call `whoami()`" ?

**Comment by chpapa** [comment]
@bensonby the inline link style of code ['getFunc()'] should have already been update in this put request. Users will know it is a clickable link by the blue color, or u think the blue color is not enough?

**Comment by chpapa** [comment]
Btw, I have made the doc generator automatically add id in h1, so if u spot any <a name=""> I think you can remove them

**Comment by bensonby** [comment]
@chpapa yes I checked that, perhaps it's due to the blue text on grey background, maybe just the style?

**Comment by chpapa** [comment]
@tensiuyan when we are done with the basic style and the two pages update, let's merge this first? I think each page's inline api ref update should have it's own PR (if not its own issue...)

**Comment by tensiuyan** [comment]
sure. thanks.

**Comment by tensiuyan** [comment]
@chpapa was trying to work on adding inline API reference. Though saw this branch has conflict. Can you help take a look?

**Pull request closed without merging**